### PR TITLE
Remove global logger from outputs and monitoring

### PR DESCRIPTION
**Pull request by simitt**
## What does this PR do?

Remove global logger for `libbeat/outputs` and `libbeat/monitoring`. 

## Why is it important?
Precondition for improved structured logging.

## Checklist

- [x] My code follows the style guidelines of this project
~- [ ] I have commented my code, particularly in hard-to-understand areas~
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
~- [ ] I have added tests that prove my fix is effective or that my feature works~

## Author's Checklist

- [x] Add dev changelog
- [x] needs to be rebased once https://github.com/elastic/beats/pull/16734 is merged

## Related issues

- Relates elastic/beats#15699

## Developer Docs

Following public methods are changed to take a logger as first parameter now:
* `common.transport.transport#ProxyDialer`
* `common.transport.tlscommon.tls#ReadPEMFile` 
* `outputs.elasticsearch.client#BulkReadItemStatus`

### CODE REVIEW

**Comment by urso** [comment]
@ycombinator I think parts of this interfere with your refactoring.

**Comment by simitt** [comment]
@urso , @ycombinator - yes they do, but I was already done with the changes when I saw the other PR. Don't worry about it, I will just rebase my PR once the other one is merged.

**Comment by ycombinator** [comment]
Thanks @urso, @simitt. Pretty sure this PR here will go in before the refactoring one, given their relative sizes. So please go ahead here and I will incorporate your changes into my larger refactoring PR.

**Comment by simitt** [comment]
Failing tests seem unrelated.

**Comment by urso** [comment]
LGTM.

Thank you for taking the time and doing so many changes at once!

**Comment by simitt** [comment]
@urso I rebased my PR on top of the changes of #16734, can you please re-review the changes from the last two commits.

File: CHANGELOG-developer.next.asciidoc [hunk 1]
```diff
@@ -33,6 +33,8 @@ The list below covers the major changes between 7.0.0-rc2 and master only.
 - `queue.Eventer` has been renamed to `queue.ACKListener` {pull}16691[16691]
 - Require logger as first parameter for `outputs.transport.transport#ProxyDialer` and `outputs.elasticsearch.client#BulkReadItemStatus`. {pull}16761[16761]
 
+- The `libbeat/outputs/transport` package has been moved to `libbeat/common/transport`. {pull}16734[16734]
+- The `libbeat/outputs/tls.go` file has been removed. All exported symbols in that file (`libbeat/outputs.*`) are now available as `libbeat/common/tlscommon.*`. {pull}16734[16734]
```
Comment (urso) [inline CHANGELOG-developer.next.asciidoc]: how come these did end up as new in this PR?

Comment (urso) [reply CHANGELOG-developer.next.asciidoc]: Nevermind, did look at merge commit :)

**Pull request merged**